internal/policy/operator/required_annotations.go

```diff
@@ -0,0 +1,91 @@
+package operator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"github.com/operator-framework/api/pkg/manifests"
+	operatorsv1alpha1 "github.com/operator-framework/api/pkg/operators/v1alpha1"
+
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/check"
+	libcsv "github.com/redhat-openshift-ecosystem/openshift-preflight/internal/csv"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/image"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/internal/log"
+)
+
+var requiredAnnotations = []string{
+	libcsv.DisconnectedAnnotation, libcsv.FipsCompliantAnnotation, libcsv.ProxyAwareAnnotation, libcsv.TLSProfilesAnnotation,
+	libcsv.TokenAuthAWSAnnotation, libcsv.TokenAuthAzureAnnotation, libcsv.TokenAuthGCPAnnotation,
+}
+
+var _ check.Check = &RequiredAnnotations{}
+
+type RequiredAnnotations struct{}
+
+func (h RequiredAnnotations) Validate(ctx context.Context, imageReference image.ImageReference) (result bool, err error) {
+	return h.validate(ctx, imageReference.ImageFSPath)
+}
+
+func (h RequiredAnnotations) getBundleCSV(_ context.Context, bundlepath string) (*operatorsv1alpha1.ClusterServiceVersion, error) {
+	bundle, err := manifests.GetBundleFromDir(bundlepath)
+	if err != nil {
+		return nil, err
+	}
+	return bundle.CSV, nil
+}
+
+func (h RequiredAnnotations) validate(ctx context.Context, bundledir string) (bool, error) {
+	logger := logr.FromContextOrDiscard(ctx)
+	csv, err := h.getBundleCSV(ctx, bundledir)
+	if err != nil {
+		return false, err
+	}
+
+	missingAnnotations := []string{}
+	incorrectValues := map[string]string{}
+	for _, annotation := range requiredAnnotations {
+		value, ok := csv.GetAnnotations()[annotation]
+		if !ok {
+			missingAnnotations = append(missingAnnotations, annotation)
+			continue
+		}
+		// the only string values allowed are lower case 'true' or 'false'
+		if !(value == "true" || value == "false") {
+			incorrectValues[annotation] = value
+		}
+	}
+
+	if len(missingAnnotations) > 0 {
+		logger.V(log.DBG).Info("expected annotations are missing", "missingAnnotations", missingAnnotations)
+	}
+
+	if len(incorrectValues) > 0 {
+		for key, value := range incorrectValues {
+			logger.V(log.DBG).Info(fmt.Sprintf("expected annotation: %s to have either 'true' or 'false' value, but had value of: %s.", key, value))
+		}
+	}
+
+	return len(missingAnnotations) == 0 && len(incorrectValues) == 0, nil
+}
+
+func (h RequiredAnnotations) Name() string {
+	return "RequiredAnnotations"
+}
+
+func (h RequiredAnnotations) Metadata() check.Metadata {
+	return check.Metadata{
+		Description: "Checks that the CSV has all of the required feature annotations.",
+		// TODO: This will start as warning, but will need to move to `best`
+		Level:            "warning",
```

Comment:
Might go with `warn` instead of `warning`. I could be convinced otherwise on this one.

Reply:
`warn` probably makes most sense, normal checks are level `best`